0.12

balloob opened the pull request:
It's been a busy two weeks so it's time to get ready for a next release: 0.12.
- Binary sensor: command sensor added (@Bart274)
- Nest support extended to include sensors and binary sensors (@joshughes)
- Toggle service added to `homeassistant`, `switch`, `light` and `media_player` (@rmkraus)
- Light: LIFX platform added (@TangoAlpha)
- Notify: Twitter platform added (@HydrelioxGitHub)
- Sensor: Template platform added (@pavoni)
- Switch: Wink platform now supports sirens (@w1ll1am23)
- Insteon hub support added (@FreekingDean)
- Statsd component added (@michaelkuty)
- Light: Rfxtrx platform now supports dimming (@turbokongen)
- Event helpers are now also available as decorators for custom components (@rmkraus)
- Thermostat: services added to control fans (@auchter)
- Frontend: support added for tabs to show different views of your house (@balloob)
- Time scheduling (including automation) now works with intervals (ie. `/5`) (@kennedyshead)
- Sensor: onewire support added (@deisi)
- Zigbee support added (@flyte)
- Device Tracker: OwnTracks can now track iBeacons (@pavoni)
- Notify: Google Voice SMS platform added (@w1ll1am23)
- Bugfixes by @molobrakos, @MartinHjelmare, @pavoni, @trollkarlen, @zmrow, @maddox, @persandstrom, @happyleavesaoc, @balloob, @fabaff, @stefan-jonasson, @haraldnagel.

![screen shot 2016-01-24 at 14 13 16](https://cloud.githubusercontent.com/assets/1444314/12574447/cae4a862-c3b8-11e5-85e5-f2fcf9c70442.png)
### Deprecations
- Nest config has moved from thermostat to the `nest` component.

auchter committed:
thermostat: move fan attribute up to thermostat

auchter committed:
thermostat: add service to control fan mode

auchter committed:
nest: implement fan control

FreekingDean committed:
Added Insteon Support

trollkarlen committed:
sonos: add hosts and interface_addr to sonos config

Config can now specify one or more hosts to connect and
also a interface_addr to multicast on if multiple interfaces exists.

Signed-off-by: Robert Marklund <robbelibobban@gmail.com>

balloob committed:
Merge pull request #936 from MartinHjelmare/fix-mysensors-sensors

Fix missing binary sensor types

balloob committed:
Merge pull request #934 from balloob/mqtt-protocol

Allow forcing MQTT protocol v3.1

molobrakos committed:
1) artwork_url might be a relative url (such as /imageproxy). in that case, join it with the base url. note: urllib.parse.urljoin will handle case when the artwork url is absolute. 2) artwork would not be replaced in the user interface because the url did not change between tracks (http://.../cover.jpg). solved by appending internal hash of the media title to the url to force reload

pavoni committed:
Remove sensor state from switch and wemo

pavoni committed:
Tidy

pavoni committed:
More tidying.

molobrakos committed:
less hacky way of getting unique cover art

balloob committed:
Merge pull request #939 from molobrakos/squeezebox-fix

artwork fixes

pavoni committed:
Fix bug related to maker_param types

pavoni committed:
Remove ghost debug code.

balloob committed:
Merge pull request #940 from balloob/remove_sensor_state

Remove sensor state

trollkarlen committed:
influxdb: fix the need of admin to run

Use select statment to show if db exits instead of
'SHOW DATABASES' which cant be run by a non admin user.
See https://github.com/influxdata/influxdb/issues/4785 for more info.

Also influxdb dont like empty writes('') so ignore state changes of that kind,
this happens on startup of home assistant.

Signed-off-by: Robert Marklund <robbelibobban@gmail.com>

HydrelioxGitHub committed:
Add Twitter as a notification platform

HydrelioxGitHub committed:
Clean some errors

balloob committed:
Merge pull request #944 from HydrelioxGitHub/twitter_notify

Add Twitter as a notification platform

balloob committed:
Merge pull request #943 from trollkarlen/influxdb-fix

influxdb: fix the need of admin to run

zmrow committed:
Fix #863 - Added a check for good pull from OWM and return if the object is None

balloob committed:
Merge pull request #945 from zmrow/fix-exception-in-openweather

Fix #863 - Added a check for good pull from OWM and return if the object is None

balloob committed:
Merge pull request #937 from trollkarlen/sonos-multicast

sonos: add hosts and interface_addr to sonos config

Commit:
Wink siren support

Commit:
Fixed style

pavoni committed:
First drafy of sensor.template.

michaelkuty committed:
Add statsd component.

pavoni committed:
Revise to not need dependencies (or quoting)!

pavoni committed:
Add test.

pavoni committed:
Merge branch 'dev' into sensor_template

trollkarlen committed:
kodi: remove traceback on state check

If the computer is sleeping that are running xbmc then
the traceback is printed in the log all the time.
Keep the traceback in debug mode of the module if needed.

Signed-off-by: Robert Marklund <robbelibobban@gmail.com>

balloob committed:
Merge pull request #953 from trollkarlen/kodi-fix

kodi: remove traceback on state check

trollkarlen committed:
yr.no: fix some bugs in the component

Someone forgot some in:s and ()
Also added windGust sensor.

Signed-off-by: Robert Marklund <robbelibobban@gmail.com>

balloob committed:
Merge pull request #957 from trollkarlen/yr.no-fix

yr.no: fix some bugs in the component

maddox committed:
let port be optional

maddox committed:
use custom material icons to represent speaker state

maddox committed:
not using these

pavoni committed:
Tidy.

pavoni committed:
Merge branch 'dev' into sensor_template

balloob committed:
Merge pull request #958 from balloob/itunes-updates

iTunes Updates

balloob committed:
Script: use config service helper

balloob committed:
Merge pull request #963 from balloob/scrip-service-helper

Script: use config service helper

balloob committed:
Update mdi icons

balloob committed:
Merge pull request #962 from FreekingDean/dev

Adding Support for Insteon Hub

balloob committed:
Allow passive zones

pavoni committed:
Tidy, add test for ValueException logic.

pavoni committed:
Handle race condition on startup.

balloob committed:
Merge pull request #950 from michaelkuty/dev

Add statsd component.

pavoni committed:
Change error state to be 'error' rather than 'unknown', trace error.

pavoni committed:
Merge pull request #948 from balloob/sensor_template

First draft of sensor.template.

HydrelioxGitHub committed:
Add icons to NetAtmo sensors

balloob committed:
Merge pull request #967 from HydrelioxGitHub/NetAtmo-icons

Add icons to NetAtmo sensors

trollkarlen committed:
smtp: make smtp component thread safe

Also fix so it does not require ip, port, username and password when using local smtp server.
Add debug config.

Signed-off-by: Robert Marklund <robbelibobban@gmail.com>

turbokongen committed:
Added support for Dimming with lights in Rfxtrx module

turbokongen committed:
Small intendation fix

turbokongen committed:
Fixing of various test errors

turbokongen committed:
flake8 complaint fix

turbokongen committed:
Another flake8 fix too long line

molobrakos committed:
only query artwork by track_id if id is available (7.7 vs 7.9 version issue?)

molobrakos committed:
style

molobrakos committed:
style

molobrakos committed:
should be _id

sdague committed:
fix typo in log message

The plex component logs an htts url, which is confusing to people, as
they think something is broken, when it is not.

Closes #959

Commit:
Separate LIFX code and HA component

Commit:
Bump version liffylights

balloob committed:
Merge pull request #977 from sdague/dev

fix typo in log message

rmkraus committed:
Created automation decorator prototype

Created an initial iteration of an Automation decorator.

Commit:
Update liffylights version

Commit:
Fixes for lint

Commit:
Import 3rd party library inside method

Commit:
Update requirements

balloob committed:
Upgrade PyChromecast version

balloob committed:
Merge pull request #973 from turbokongen/dev

Added support for Dimming with lights in Rfxtrx module

balloob committed:
Merge pull request #971 from trollkarlen/smtp-fix

smtp: make smtp component thread safe

balloob committed:
Merge pull request #964 from balloob/passive-zones

Allow passive zones

balloob committed:
Merge pull request #980 from balloob/upgrade-cast

Upgrade PyChromecast version

Commit:
Add to .coveragerc

balloob committed:
Load YAML config into an ordered dict

balloob committed:
Move split_entity_id to helpers

balloob committed:
Move extract_entity_id to service helpers

balloob committed:
Move generate_entity_id to entity helpers

balloob committed:
Merge pull request #981 from balloob/ordered-yaml

Load YAML config into an ordered dict

flyte committed:
Add zigbee components.

Commit:
Updated the entity names for all z-wave devices.
Z-Wave have a unique node id, this is now added to the entity name.

This is a change will break old configs.  The Z-Wave entity_id will have to be updated in configurations.

The reason for this change is that the old behavior created duplicate entity_ids if you have several similar devices. The order devices is included is not consistent so when the service was restarted the entity ids could change and the configuration for entities will be invalid.

There was also inconsistencies between the naming for sensors and switches.
(cherry picked from commit 33a78df)

turbokongen committed:
Merge pull request #1 from balloob/dev

Updating base

Commit:
Fix wrongly generated requirements

Commit:
Fix wrongly generated requirements

turbokongen committed:
Send correct command to pyrfxtrx

Although it seems to work with send_on, it throws an logged error. Using correct command against pyrfxtrx removes this error.

turbokongen committed:
flake8 complaint fix

balloob committed:
Merge pull request #986 from turbokongen/dev

Small errorlog fix.

balloob committed:
Update version pynetgear

balloob committed:
Merge pull request #987 from balloob/update-pynetgear

Update version pynetgear

auchter committed:
thermostat: split up services

rmkraus committed:
Reverting Automation decorator in favor of a new approach.

rmkraus committed:
Moved sunrise/sunset tracking to helpers

The automation component contained some very handy and generic
functions for tracking sunset and sunrise. This was moved to
helpers/event.py.

rmkraus committed:
Initial pass at event decorators

Created event decorators for custom components. Decorators were created
for the events: track_state_change, track_sunrise, track_sunset, and
track_time_change.

rmkraus committed:
Fixed bugs to allow HA to boot again

1) helpers/event should not import the sun component unless it is
requested. This prevents circular import.
2) fixed import typo in bootstrap
2) bootstrap cannot import event_decorators until it is needed because
this leads to a circular import.

molobrakos committed:
Merge pull request #990 from molobrakos/squeezebox-fix

bugfix for squeezebox cover art regression

rmkraus committed:
A few lint fixes to event decorators.

balloob committed:
Allow groups to be used as views

rmkraus committed:
Another revision on event decorators

This revision of event decorators removes much of the complexity. The
decorated functions are no longer wrapped with a class that tracks
last_run, etc. Bootstrap now gives hass to the event_decorators module
before initializing components so the decorators no longer require
activation.

rmkraus committed:
Many updates regarding event decorators

1. Added HASS to the arguments for callbacks that are created with
event decorators.
2. Added a service decorator.
3. Updated example.py in the example config to use the event decorators.

rmkraus committed:
Updated example.py component

Cleaned up example.py to better handle failed loads.

rmkraus committed:
Lint fixes and additions to event decorators

1. service decorator was overwriting the function name with one of its
arguments.
2. Accidentally left an extra argument in track_sunrise.
3. Added track_utc_time_change decorator.

rmkraus committed:
Added event helper tests

1. Added tests for all event decorators
2. Added tests for sunrise and sunset event helpers

Commit:
Add preliminary support for transition time

Commit:
Bump version of liffylights

Commit:
Fix long line

rmkraus committed:
Moved service decorator to service helpers

Moved the service decorator to the service helpers module and moved the
associated tests.

rmkraus committed:
Merge remote-tracking branch 'balloob/dev' into automation-decorator

# Conflicts:
#	homeassistant/helpers/service.py
#	tests/helpers/test_service.py

rmkraus committed:
Fixed lint issue from merge

extract_entity_ids from the service helpers was overwriting the service
decorator with one of its attributes. This was fixed.

rmkraus committed:
Removed service decorator from event decorators

rmkraus committed:
Removed decorator callback

The decorator callback was not actually necessary so it was removed and
replaced with a partial function instead.

balloob committed:
Merge pull request #979 from rmkraus/automation-decorator

Automation Decorator for custom components

balloob committed:
Merge pull request #992 from balloob/sections

Add support for sections in the frontend [Fixes #100705168]

balloob committed:
Merge pull request #988 from auchter/thermostat-fan

Add service to control a thermostat's fan

Commit:
Update requirements_all

balloob committed:
Hide groups when they are views or auto defined

Commit:
Update to v0.6 of liffylights

Commit:
Update requirements_all

Commit:
Update to v0.8.3 of liffylights

Commit:
Update requirements_all

Commit:
Add extra debugging
Add duration to power on

Commit:
Fix liffylights import

Commit:
Update requirements_all

Commit:
Update for new liffylights version

persandstrom committed:
update vsure version

Commit:
Update liffylights version

Commit:
Fix logging typo

persandstrom committed:
Merge pull request #995 from persandstrom/vsure0.4.8

update python-verisure to version 0.4.8

Commit:
Brightness from HA overrides brightness from HSV conversion

Commit:
bump orvibo version

MartinHjelmare committed:
Fix unit of measurement

* Add version check in unit_of_measurement, to avoid error.
* Remove not needed metric check in unit_of_measurement.
* Add more value types to unit_of_measurement.
* Fix unit prefix
* Remove unused SCAN_INTERVAL variable.

happyleavesaoc committed:
Merge pull request #994 from happyleavesaoc/orvibo_bump

Orvibo version bump

balloob committed:
Merge pull request #930 from TangoAlpha/dev

Add LIFX bulb support

balloob committed:
Update frontend

MartinHjelmare committed:
Fix adding devices on the fly

Devices were not added without persistence enabled and restart of HA.
Node id was added to defaultdict(list) by mistake when checking if list
of defaultdict was empty.

* Fix adding devices in mysensors_callback.
* Change devices to regular dict.

balloob committed:
Add views to demo

balloob committed:
Fix lint issue

balloob committed:
Update frontend with view url sync

balloob committed:
Merge pull request #1000 from balloob/view-url-sync

Update frontend with view url sync

balloob committed:
Merge branch 'master' into dev

Conflicts:
	homeassistant/const.py

balloob committed:
Fix Chrome url sync

fabaff committed:
Update docstring

Commit:
Updated teh entity_id generation

Entity ids will now look like this:
sensor.philio_technology_corporation_psm021_slim_multisensor_doorwindow_sensor_7
sensor.philio_technology_corporation_psm021_slim_multisensor_luminance_7
sensor.philio_technology_corporation_psm021_slim_multisensor_motion_sensor_7
sensor.philio_technology_corporation_psm021_slim_multisensor_tamper_sensor_7
sensor.philio_technology_corporation_psm021_slim_multisensor_temperature_7

sensor.philio_technology_corporation_psm021_slim_multisensor_doorwindow_sensor_8
sensor.philio_technology_corporation_psm021_slim_multisensor_luminance_8
sensor.philio_technology_corporation_psm021_slim_multisensor_motion_sensor_8
sensor.philio_technology_corporation_psm021_slim_multisensor_tamper_sensor_8
sensor.philio_technology_corporation_psm021_slim_multisensor_temperature_8

And if there is several values of the same type in a node the instance id will be appended like this (three switches in the same device):
switch.fibaro_system_unknown_type0202_id1002_switch_9
switch.fibaro_system_unknown_type0202_id1002_switch_9_2
switch.fibaro_system_unknown_type0202_id1002_switch_9_3

MagnusKnutas committed:
Running periodic tasks

MagnusKnutas committed:
hourly periodic task

MagnusKnutas committed:
test past midnight support for hours

MagnusKnutas committed:
Warnings for not divisible settings

MagnusKnutas committed:
Some warnings if input is wrong for periodic tasks

MagnusKnutas committed:
New and improved handling of the matching! Kudos to @balloob

MagnusKnutas committed:
Year periodic test (just to trigger travis but should maybe be here anyway)

MagnusKnutas committed:
Unused import was the problem

MagnusKnutas committed:
Remove warnings no longer needed (I think :D)

MartinHjelmare committed:
Merge pull request #999 from MartinHjelmare/fix-not-adding-devices

Fix adding devices on the fly

MagnusKnutas committed:
Sorry, I was wrong they are needed

MagnusKnutas committed:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	homeassistant/components/light/zwave.py
#	homeassistant/components/switch/zwave.py

MagnusKnutas committed:
Fixes bug in time trigger and adds test for ex. /two

MagnusKnutas committed:
pushed test code :) fixed now

MagnusKnutas committed:
Tests for time and a fix for wrong access of _LOGGER

MagnusKnutas committed:
And a test just to see if nothing works if not providing any vars

MagnusKnutas committed:
Remove not needed log warning

MagnusKnutas committed:
Dont wrap '/int' parameters with tuple

MagnusKnutas committed:
Remove unused import

Commit:
Moved common z-wave node functionality to a common z-wave class

fabaff committed:
Updgrade to astral 0.9

fabaff committed:
Upgrade influxdb to 2.11.0

fabaff committed:
Upgrade python-telegram-bot to 3.2.0

fabaff committed:
Upgrade blockchain to 1.2.1

fabaff committed:
Update docstrings

MartinHjelmare committed:
Fix device_state_attributes

* Return a dict with keys as the names of the enum members instead of
	the member values.
* This fixes a TypeError: unorderable types: int() < str()

rmkraus commented:
> Insteon support added (@FreekingDean)

Should be Insteon *Hub* support added. Other than that, I can't believe how busy it has been!

MartinHjelmare committed:
Merge pull request #1010 from MartinHjelmare/fix-device-attr

Fix device_state_attributes

balloob committed:
Merge pull request #1009 from fabaff/docstrings

Update docstrings

balloob committed:
Merge pull request #1008 from fabaff/blockchain-upgrade

Upgrade blockchain to 1.2.1

balloob committed:
Merge pull request #1007 from fabaff/telegram-upgrade

Upgrade python-telegram-bot to 3.2.0

balloob committed:
Merge pull request #1006 from fabaff/influxdb-upgrade

Upgrade influxdb to 2.11.0

balloob committed:
Merge pull request #1005 from fabaff/astral-upgrade

Updgrade to astral 0.9

MartinHjelmare committed:
Merge pull request #1011 from MartinHjelmare/fix-unit-of-measurement

Fix unit of measurement

balloob committed:
Merge branch 'pr/78' into dev

Conflicts:
	homeassistant/const.py

balloob committed:
Merge pull request #1002 from kennedyshead/dev

Running periodic tasks

fabaff committed:
Add link to docs and update docstrings

Commit:
Updated Imports.
ZWave modules now import the domain from the base domain.

balloob committed:
Merge pull request #982 from stefan-jonasson/dev

Updated the entity names for all z-wave devices.

fabaff committed:
Add twitter docs

balloob committed:
Add onewire to coverage

fabaff committed:
Add link to docs and remove configuration details

fabaff committed:
Add link to docs

fabaff committed:
Update docstring

fabaff committed:
Add link to docs

fabaff committed:
Update docstring and fix typos

fabaff committed:
Move configuration details to docs

fabaff committed:
Add link to docs

fabaff committed:
Update docstings

fabaff committed:
Upgrade py-cpuinfo to 0.1.8

tomduijf commented:
:+1: 
Busy indeed, mailbox exploded these last couple of weeks, awesome progress!

fabaff committed:
Upgrade psutil to 3.4.2

MartinHjelmare committed:
Refactor unit of measurement

* Add dict with map of units to v_types.
* Avoid numerous if, elif branches and return statements.
* Add custom unit decoupled from other units.

balloob committed:
Merge pull request #1016 from fabaff/cpuinfo-upgrade

Upgrade py-cpuinfo to 0.1.8

balloob committed:
Merge pull request #1017 from fabaff/psutil-upgrade

Upgrade psutil to 3.4.2

MagnusKnutas committed:
Logbook service

MagnusKnutas committed:
Cleanup

MagnusKnutas committed:
And with template support

MagnusKnutas committed:
Less complex service setup

MartinHjelmare committed:
Refactor s_types and v_types mapping

To be able to handle new platforms with multiple used v_types but
single states, a new more strict mapping was required. Each s_type is
now mapped to its corresponding v_type(s) in a dict of lists.

Commit:
Move Insteon API KEY into configuration.yaml

Commit:
Use validate_config, watch for import exception

MagnusKnutas committed:
logbook.log the name of the game

flyte committed:
Fix import-before-install of xbee-helper dependency.

molobrakos committed:
allow monitoring weather at other location than home (such as summer house)

balloob committed:
Merge pull request #1025 from molobrakos/yr-position

allow monitoring weather at other location than home

Commit:
pygooglevoice-sms support

Commit:
pygooglevoice-sms support

Commit:
Updated googlevoicesms version

Commit:
Remove unnecessary try around import

Commit:
Add ability to control dimmable sources

balloob committed:
Merge pull request #1022 from haraldnagel/dev

Move Insteon API KEY into configuration.yaml

balloob committed:
Fix imports

balloob committed:
Merge pull request #1035 from balloob/fix-imports

Fix imports and rename insteon to insteon_hub

MagnusKnutas committed:
Added entity_id and domain

pavoni committed:
Improve owntracks events, add support for stationary ibeacons.

pavoni committed:
Merge pull request #1036 from balloob/owntracks_beacon2

Improve owntracks events, add support for stationary ibeacons.

flyte committed:
ZigBee: Handle case in which Serial port is unable to open

flyte committed:
ZigBee: Remove extraneous update_ha_state() usage and use it more appropriately for initial state update.

pavoni committed:
First cut mobile beacon tracking.

pavoni committed:
Add docstring, remove trace.

flyte committed:
ZigBee: Ensure correct entity types are used for each component.

pavoni committed:
Revise beacon name. Add debug.

pavoni committed:
Revise tracked beacon name.

pavoni committed:
Copy kwargs, remove traces.

flyte committed:
Catch ZigBee exceptions when communicating with the devices and log appropriate messages.

pavoni committed:
Fix typo.

balloob committed:
Merge pull request #1023 from flyte/zigbee-pr

Add ZigBee integration and components.

balloob committed:
Merge pull request #1020 from kennedyshead/dev

Adding service for logbook

MartinHjelmare committed:
Merge pull request #1033 from MartinHjelmare/refactor-unit-of-measurement

Mysensors: Refactor unit of measurement

pavoni committed:
Remove dev/null from build script.

Commit:
Added target support for googlevoice

Commit:
Fixed style errors

Commit:
Added test exclude in .coveragerc

MartinHjelmare committed:
Merge pull request #1034 from MartinHjelmare/refactor-s_v_types

Mysensors: Refactor s_types and v_types mapping

balloob committed:
Merge pull request #1037 from balloob/owntracks_mobile_beacons

Owntracks - mobile ibeacon tracking.

balloob committed:
Remove owntracks from coverage

Commit:
Add tests for color util

Commit:
Fixed style error

Commit:
Validate target

Commit:
Log if target == NULL

balloob committed:
Merge pull request #1044 from w1ll1am23/googlevoice

Google Voice SMS notification support

Commit:
Fix #1046 by correcting platform name

balloob committed:
Merge pull request #1047 from haraldnagel/dev

Fix #1046 by correcting platform name

balloob committed:
Version bump to 0.12
